aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -1334,8 +1334,23 @@ tensorInfoLegacyIfScalar(cuda::detail::TensorInfo<T, IndexType> ti) {
   return ti;
 }
 
+uint64_t getDefaultMaxThreadsPerBlock() {
+#ifndef USE_ROCM
+  return 128;
+#else
+  hipDeviceProp_t* prop = at::cuda::getCurrentDeviceProperties();
+  // MI300x optimization
+  if (std::string(prop->gcnArchName).find("gfx94") != std::string::npos) {
```

Comment:
I'm a bit worried about the future generations, which I'm sure we'll forget about this and fall back to 256. I wonder if we can just use 1 value for rocm like 256 or 512? What's the impact of using 512/1024 for MI300-?

Reply:
> I'm a bit worried about the future generations, which I'm sure we'll forget about this and fall back to 256. I wonder if we can just use 1 value for rocm like 256 or 512? What's the impact of using 512/1024 for MI300-?

Using 512 or 1024 is fine for MI300 series. Perf wise, using 1024 is better than using 512 which is better than using 256, in this context. 
Using just 1 value of 512 is fine as the default is fine.  (256 is more conservative and is also much better than using 128).